docs/api/apiv3/paths/storage.yml

```diff
@@ -22,7 +22,64 @@ get:
       content:
         application/hal+json:
           schema:
-            $ref: '../components/schemas/storage_model.yml'
+            $ref: '../components/schemas/storage_read_model.yml'
+    '404':
+      content:
+        application/hal+json:
+          schema:
+            $ref: '../components/schemas/error_response.yml'
+          example:
+            _type: Error
+            errorIdentifier: urn:openproject-org:api:v3:errors:NotFound
+            message: The requested resource could not be found.
+      description: |-
+        Returned if the storage does not exist or the client does not have sufficient permissions to see it.
+
+        **Required permission:** view file links
+        
+put:
+  summary: Updates a storage.
+  operationId: update_storage
+  tags:
+    - File links
+  description: |-
+    Updates a storage resource. Only data that is not generated by the server can be updated. This excludes the OAuth 2
+    application data.
+  parameters:
+    - name: id
+      description: Storage id
+      in: path
+      required: true
+      schema:
+        type: integer
+      example: 1337
+  requestBody:
+    content:
+      application/json:
+        schema:
+          $ref: '../components/schemas/storage_write_model.yml'
+  responses:
+    '200':
+      description: OK
+      content:
+        application/hal+json:
+          schema:
+            $ref: '../components/schemas/storage_read_model.yml'
+    '403':
+      content:
+        application/hal+json:
+          schema:
+            $ref: '../components/schemas/error_response.yml'
+          examples:
+            response:
+              value:
+                _type: Error
+                errorIdentifier: urn:openproject-org:api:v3:errors:MissingPermission
+                message: You are not authorized to access this resource.
+      description: |-
+        Returned if the client does not have sufficient permissions.
+
+        **Required permission:** admin
```

Comment:
The permission stated to be required for the `404` is incorrect. In fact, it can be deleted completely and the description be shortened to just say: 

'Returned if the storage does not exist.'

Reply:
I disagree, if a storage exists, but the user does not have the permission to view file links, he should see a `404` when trying to patch or delete this storage.